XenAdmin/SettingsPanels/SnmpEditPage.cs

```diff
@@ -0,0 +1,278 @@
+﻿/* Copyright (c) Cloud Software Group, Inc. 
+ * 
+ * Redistribution and use in source and binary forms, 
+ * with or without modification, are permitted provided 
+ * that the following conditions are met: 
+ * 
+ * *   Redistributions of source code must retain the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer. 
+ * *   Redistributions in binary form must reproduce the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer in the documentation and/or other 
+ *     materials provided with the distribution. 
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF 
+ * SUCH DAMAGE.
+ */
+
+using System;
+using System.Drawing;
+using System.Windows.Forms;
+using XenAPI;
+using XenAdmin.Actions;
+using XenAdmin.Core;
+using System.Text.RegularExpressions;
+using XenAdmin.Actions.SNMP;
+
+namespace XenAdmin.SettingsPanels
+{
+    public partial class SnmpEditPage : UserControl, IEditPage
+    {
+        public string SubText => Messages.SNMP_EDIT_PAGE_TEXT;

```

Comment:
Very trivial thing, but since you'll be making changes in this file anyway let's tidy up a bit:
- Sort usings alphabetically.
- Move the constructor after the private fields, and SubText and Image properties after the constructor.
- Move SaveSettings closer to the other IEditPage implementation methods.
- Leave a line between all methods so the code can breath a bit (lack of lines is normally a C++ style).